scan/lib/scan/module.rb

```diff
@@ -24,7 +24,7 @@ def scanfile_name
 
     def building_mac_catalyst_for_mac?
       return false unless Scan.project
-      Scan.project.supports_mac_catalyst? && Scan.config[:catalyst_platform] == "macos"
+      Scan.config[:catalyst_platform] == "macos" && Scan.project.supports_mac_catalyst?
```

Comment:
Changing order here allows to skip project chekc in case current platform is not `macos`